### PR TITLE
Content Card Customization Example: Carousel 

### PR DESCRIPTION
**Pull request by josh-mccrowell-braze**
This PR is to review the `carousel_view` article for the Content Card documentation refresh. 

In our current examples of building this customization (for iOS and Android) we don't provide any sample code. (This article does not exist in the Web SDK documentation.) I think it would be great to provide sample code for a custom view controller. This would make these example use cases more valuable and would help us address some feedback we've received about our docs. 

Can I get a technical resource help me build out the placeholder text for each SDK in each section? 

**Open Questions**
Both the [Android article](https://www.braze.com/docs/developer_guide/platform_integration_guides/android/content_cards/customization/use_cases/carousel_view) and Swift article currently call out a view controller in Step 1, but I didn't know that view controllers were a thing in the Android context. Am I remembering wrong?

Is this terminology also appropriate for the Web context?

**Web-specific Questions**
This article was written without a Web SDK version to pull from, so I have no idea if the steps make sense for web developers. Can you please provide guidance?

### CODE REVIEW

**Comment by jerielng** [comment]
> I didn't know that view controllers were a thing in the Android context. Am I remembering wrong?

View controllers can be used as a generic term, but I think in this context, it might make sense to say `fragment`, maybe?

File: _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md [hunk 1]
```diff
@@ -1,58 +1,111 @@
...
+With this approach, you will not use the default views and logic but instead display the Content Cards in a completely custom manner by using your own views populated with data from the Braze models.
 
-In terms of the level of development effort, the key differences between the default implementation and the carousel implementation include:
-
-- Building your own views
-- Logging Content Card analytics
-- Introducing additional client-side logic to dictate how many and which cards to show in the carousel
+To implement a carousel view, you will:
+1. Build your own view controller
+2. Subscribe to your custom view controller
```
Comment (bryanlogan) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:20]: "Subscribe to your custom view" seems odd.  You might have callbacks, not subscriptions.  But isn't this really for the next step, so maybe they can be combined.

"Setup the code needed in your view to log card analytics like display, click, etc".

File: _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md [hunk 2]
```diff
@@ -1,58 +1,111 @@
...
+3. Manually log analytics
+4. Introduce additional client-side logic to dictate how many and which cards to show in the carousel
 
 ## Step 1: Create a custom view controller
 
-To create the Content Cards carousel, create your own custom view controller component (such as `UICollectionViewController`) and [subscribe to data updates]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/content_cards/integration/#getting-the-data).
+{% tabs %}
+{% tab Android %}
+
+First, create your own custom view controller component. Note that you won't be able to use Braze's default [`ContentCardFragment`](https://braze-inc.github.io/braze-android-sdk/kdoc/braze-android-sdk/com.braze.ui.contentcards/-content-cards-fragment/index.html), as it's only able to handle our default Content Card types.
```
Comment (bryanlogan) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:29]: Yeah, say "View" or "Fragement" here.

Comment (bryanlogan) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:29]: Seems like telling them they can't use ContentCardFragment is weird. It's a great starting point.  You're simply changing out the RecyclerView and adapter for something else.  But a lot of the concepts remain.

File: _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md [hunk 3]
```diff
@@ -1,58 +1,111 @@
...
+INSERT SAMPLE WEB CODE HERE
+
+{% endtab %}
+{% endtabs %}
+
+
+## Step 2: Subscribe to data updates
 
-## Step 2: Implement analytics
+Then, [subscribe to data updates]({{site.baseurl}}/developer_guide/customization_guides/content_cards/logging_analytics/#listening-for-card-updates) from your custom view controller.
```
Comment (jerielng) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:53]: I can't seem to find this page. Not sure if it exists, but at least for iOS, you would do this by calling the `subscribeToUpdates` function or maintaining a reference to the `AsyncStream`. Link [here](https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/content_cards/integration/#content-cards-data-model).

File: _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md [hunk 4]
```diff
@@ -1,58 +1,111 @@
...
+
+{% endtab %}
+{% tab Web %}
+
+INSERT SAMPLE WEB CODE HERE
+
+{% endtab %}
+{% endtabs %}
+
+## Step 4: Create a Content Card observer
```
Comment (jerielng) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:81]: I think steps 2 and 4 are the same. When you subscribe to updates, you would tell your collection view to observe that subscription, so similar to what @bryanlogan mentioned in the Android portion, I think we could combine these two into the same step, unless I'm missing something about the idea of step 2? Happy to dive deeper into it!

File: _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md [hunk 5]
```diff
@@ -1,58 +1,111 @@
...
+{% tabs %}
+{% tab Android %}
 
-Implement logic that observes for [changes in your Content Cards]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/content_cards/integration/#refreshing-content-cards), handles Content Card arrival, and displays a specific number of cards in the carousel at any one time. By default, Content Cards are sorted by created date (newest first), and a user sees all cards they are eligible for.
+INSERT SAMPLE ANDROID CODE HERE
+
+{% endtab %}
+{% tab iOS %}
+
+INSERT SAMPLE SWIFT CODE HERE
```
Comment (jerielng) [inline _docs/_developer_guide/customization_guides/content_cards/examples/carousel_view.md:71]: For this piece, we could share something similar to what we have in the default view controller in BrazeUI:

Clicks:
* [Method to log the analytics methods on a card](https://github.com/braze-inc/braze-swift-sdk/blob/fad25aabd5849df5807d1c38d6de71b4b09b8e49/Sources/BrazeUI/ContentCardUI/ContentCardUIViewController.swift#L488-L501)
* [Calling that logging method in the table view controller](https://github.com/braze-inc/braze-swift-sdk/blob/fad25aabd5849df5807d1c38d6de71b4b09b8e49/Sources/BrazeUI/ContentCardUI/ContentCardUIViewController.swift#L377-L382)

Impressions:
* [Logging method](https://github.com/braze-inc/braze-swift-sdk/blob/fad25aabd5849df5807d1c38d6de71b4b09b8e49/Sources/BrazeUI/ContentCardUI/ContentCardUIViewController.swift#L377-L382)
* Calling this method is a bit tricky. We use a lot of custom logic in our BrazeUI example. I'm not sure if we want to call this out or let them define the logic themselves

Dismissal:
* [Logging method](https://github.com/braze-inc/braze-swift-sdk/blob/fad25aabd5849df5807d1c38d6de71b4b09b8e49/Sources/BrazeUI/ContentCardUI/ContentCardUIViewController.swift#L391-L398)
* [Calling the method](https://github.com/braze-inc/braze-swift-sdk/blob/fad25aabd5849df5807d1c38d6de71b4b09b8e49/Sources/BrazeUI/ContentCardUI/ContentCardUIViewController.swift#L451-L457)

**Comment by josh-mccrowell-braze** [comment]
Got some good feedback about the scope of this article. I'm going to re-cast the example use cases section into something less specific and more generic based on this feedback. Thank you all for the feedback--it will be folded into the next iteration. Closing this PR.

**Pull request closed without merging**